dgraphtest/acl_cluster.go

```diff
@@ -0,0 +1,398 @@
+	addGroup := `
+	mutation addGroup($name: String!) {
+		addGroup(input: [{name: $name}]) {
+			group {
+				name
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addGroup,
+		Variables: map[string]interface{}{
+			"name": name,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) AddRulesToGroup(group string, rules []AclGrpRules) error {
+	addRuleToGroup := `mutation updateGroup($name: String!, $rules: [RuleRef!]!) {
+		updateGroup(input: {
+			filter: {
+				name: {
+					eq: $name
+				}
+			},
+			set: {
+				rules: $rules
+			}
+		}) {
+			group {
+				name
+				rules {
+					predicate
+					permission
+				}
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addRuleToGroup,
+		Variables: map[string]interface{}{
+			"name":  group,
+			"rules": rules,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) AddToGroup(userName, group string) error {
+	addUserToGroup := `mutation updateUser($name: String!, $group: String!) {
+		updateUser(input: {
+			filter: {
+				name: {
+					eq: $name
+				}
+			},
+			set: {
+				groups: [
+					{ name: $group }
+				]
+			}
+		}) {
+			user {
+				name
+				groups {
+					name
+				}
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addUserToGroup,
+		Variables: map[string]interface{}{
+			"name":  userName,
+			"group": group,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RemoveUserFromGroup(userName, groupName string) error {
+	removeUserGroups := `mutation updateUser($name: String!, $groupName: String!) {
+			updateUser(input: {
+				filter: {
+					name: {
+						eq: $name
+					}
+				},
+				remove: {
+					groups: [{ name: $groupName }]
+				}
+			}) {
+				user {
+					name
+					groups {
+						name
+					}
+				}
+			}
+		}`
+
+	params := GraphQLParams{
+		Query: removeUserGroups,
+		Variables: map[string]interface{}{
+			"name":      userName,
+			"groupName": groupName,
+		},
+	}
+
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RemoveRuleFromGroup(group string, rulePredicate string) error {
+	removeRuleFromGroup := `mutation updateGroup($name: String!, $rules: [String!]!) {
+		updateGroup(input: {
+			filter: {
+				name: {
+					eq: $name
+				}
+			},
+			remove: {
+				rules: $rules
+			}
+		}) {
+			group {
+				name
+				rules {
+					predicate
+					permission
+				}
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: removeRuleFromGroup,
+		Variables: map[string]interface{}{
+			"name":  group,
+			"rules": []string{rulePredicate},
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) DeleteGroup(name string) error {
+	delGroup := `
```

Comment:
in some places, you write the mutation in the same line, and in some places in the next line. Why?